temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1637,11 +1637,13 @@ private void addActivityRetryTimer(RequestContext ctx, StateMachine<ActivityTask
           if (activity.getState() != State.INITIATED && data.getAttempt() != attempt) {
             return;
           }
-          selfAdvancingTimer.lockTimeSkipping(
-              "activityRetryTimer " + activity.getData().scheduledEvent.getActivityId());
+          LockHandle lockHandle =
+              selfAdvancingTimer.lockTimeSkipping(
+                  "activityRetryTimer " + activity.getData().scheduledEvent.getActivityId());
           boolean unlockTimer = false;
           try {
             update(ctx1 -> ctx1.addActivityTask(data.activityTask));
+            // shouldn't we unlock timer here too to avoid the leak?
```

Comment:
Sure seems like it, otherwise that lock is dropped forever. Can't imagine why that would be a good thing.

Reply:
Unless this callback is invoked repeatedly